### PR TITLE
Various documentation

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/data/decoration/tiles.h [hunk 1]
```diff
@@ -0,0 +1,643 @@
+#include "constants/event_objects.h"
+
```
Comment (jiangzhengwenjz) [inline src/data/decoration/tiles.h]: global symbol should start with `g`

File: src/map_name_popup.c [hunk 2]
```diff
@@ -18,106 +18,116 @@ static u16 MapNamePopupCreateWindow(bool32 palIntoFadedBuffer);
...
+            gTasks[taskId].tPalIntoFadedBuffer = palIntoFadedBuffer;
         }
         else
         {
-            if (gTasks[taskId].data[0] != 4)
-                gTasks[taskId].data[0] = 4;
-            gTasks[taskId].data[3] = 1;
+            if (gTasks[taskId].tState != 4)
+                gTasks[taskId].tState = 4;
+            gTasks[taskId].tReshow = 1;
```
Comment (jiangzhengwenjz) [inline src/map_name_popup.c]: `TRUE`

File: src/map_name_popup.c [hunk 3]
```diff
@@ -18,106 +18,116 @@ static u16 MapNamePopupCreateWindow(bool32 palIntoFadedBuffer);
...
+            if (task->tReshow)
             {
-                MapNamePopupPrintMapNameOnWindow(task->data[4]);
-                CopyWindowToVram(task->data[4], 2);
-                task->data[0] = 1;
-                task->data[3] = 0;
+                MapNamePopupPrintMapNameOnWindow(task->tWindowId);
+                CopyWindowToVram(task->tWindowId, 2);
+                task->tState = 1;
+                task->tReshow = 0;
```
Comment (jiangzhengwenjz) [inline src/map_name_popup.c]: `FALSE`

File: src/map_name_popup.c [hunk 4]
```diff
@@ -18,106 +18,116 @@ static u16 MapNamePopupCreateWindow(bool32 palIntoFadedBuffer);
...
         {
-            if (task->data[5])
+            if (task->tWindowExists)
             {
-                RemoveWindow(task->data[4]);
-                task->data[5] = 0;
-                task->data[7] = 1;
+                RemoveWindow(task->tWindowId);
+                task->tWindowExists = FALSE;
+                task->tWindowDestroyed = 1;
```
Comment (jiangzhengwenjz) [inline src/map_name_popup.c]: `TRUE`

File: src/option_menu.c [hunk 5]
```diff
@@ -566,12 +569,12 @@ static void LoadOptionMenuItemNames(void)
...
 {
-    u16 v1, v2;
+    u16  maxLetterHeight, y;
     
-    v1 = GetFontAttribute(2, FONTATTR_MAX_LETTER_HEIGHT);
-    v2 = selection * (v1 - 1) + 0x3A;
-    SetGpuReg(REG_OFFSET_WIN0V, WIN_RANGE(v2, v2 + v1));
+     maxLetterHeight = GetFontAttribute(2, FONTATTR_MAX_LETTER_HEIGHT);
+    y = selection * ( maxLetterHeight - 1) + 0x3A;
+    SetGpuReg(REG_OFFSET_WIN0V, WIN_RANGE(y, y +  maxLetterHeight));
```
Comment (GriffinRichards) [inline src/option_menu.c]: Accidental leading space in the F&R for `maxLetterHeight`

File: src/data/decoration/tiles.h (hunk 1 above)
Comment (PikalaxALT) [reply src/data/decoration/tiles.h]: Would rather keep this synced with RSE

File: src/intro.c [hunk 6]
```diff
@@ -448,58 +439,58 @@ static const union AnimCmd *const gAnimCmdTable_840BEC0[] = {
...
-	{gUnknown_8409A1C, 8},
-	{gUnknown_840B834, 10},
-	{gUnknown_840B854, 11},
+static const struct SpritePalette sFightSceneSpritePalettes[] = {
+	{sSpritePals_Gengar, 6},
+	{sSpritePals_Nidorino, 7},
+	{sSpritePals_Grass, 8},
+	{sSpritePals_GengarSwipe, 10},
+	{sSpritePals_NidorinoRecoilDust, 11},
     // {0}
```
Comment (jiangzhengwenjz) [inline src/intro.c:465]: we can use the `BUGFIX` macro defined last time

File: src/intro.c [hunk 7]
```diff
@@ -516,26 +507,26 @@ static bool8 sub_80EC62C(void)
         DmaFill32(3, 0, OAM, OAM_SIZE);
         DmaFill16(3, 0, PLTT + sizeof(vu16), PLTT_SIZE - sizeof(vu16));
         ResetPaletteFade();
-        load_copyright_graphics(0, 0x3800, 0);
+        load_copyright_graphics(0 * BG_CHAR_SIZE, 7 * BG_SCREEN_SIZE, 0);
         ScanlineEffect_Stop();
         ResetTasks();
         ResetSpriteData();
         FreeAllSpritePalettes();BeginNormalPaletteFade(0xFFFFFFFF, 0, 16, 0, 0xFFFF);
```
Comment (jiangzhengwenjz) [inline src/intro.c:514]: newline helps reading here

File: src/intro.c [hunk 8]
```diff
@@ -606,49 +598,49 @@ static void sub_80EC870(void)
         ResetSpriteData();
         ResetPaletteFade();
         ResetTempTileDataBuffers();
-        sub_80ECA00();
+        Intro_ResetGpuRegs();
         DmaFill16(3, 0, VRAM, VRAM_SIZE);
         DmaFill32(3, 0, OAM, OAM_SIZE);
         DmaFill16(3, 0, PLTT, PLTT_SIZE);
         FillPalette(0, 0, 0x400);
         ResetBgsAndClearDma3BusyFlags(0);
```
Comment (jiangzhengwenjz) [inline src/intro.c:606]: `FALSE`

File: src/intro.c [hunk 9]
```diff
@@ -606,49 +598,49 @@ static void sub_80EC870(void)
         ResetSpriteData();
         ResetPaletteFade();
         ResetTempTileDataBuffers();
-        sub_80ECA00();
+        Intro_ResetGpuRegs();
         DmaFill16(3, 0, VRAM, VRAM_SIZE);
         DmaFill32(3, 0, OAM, OAM_SIZE);
         DmaFill16(3, 0, PLTT, PLTT_SIZE);
         FillPalette(0, 0, 0x400);
```
Comment (jiangzhengwenjz) [inline src/intro.c:605]: `RGB_BLACK`

File: src/intro.c [hunk 10]
```diff
@@ -665,286 +657,286 @@ static void sub_80ECA00(void)
...
     {
     case 0:
-        InitWindows(gUnknown_840BBA8);
-        LZ77UnCompWram(gUnknown_840272C, this->field_043C);
-        LZ77UnCompWram(gUnknown_84028F8, this->field_003C);
+        InitWindows(sWindowTemplate);
+        LZ77UnCompWram(sBlit_GameFreakText, this->gamefreakTextBitmap);
+        LZ77UnCompWram(sSpriteTiles_GameFreakLogoArt, this->gamefreakLogoArtSpriteTiles);
         FillBgTilemapBufferRect(2, 0x000, 0, 0, 32, 32, 0x11);
         FillWindowPixelBuffer(0, 0x00);
```
Comment (jiangzhengwenjz) [inline src/intro.c:691]: `PIXEL_FILL(0)`

**Pull request merged**